From 8e8e2cfc0808d3bda9b7c73b9e7b2c15eccf049c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Wed, 5 Nov 2025 20:53:22 +0100
Subject: [PATCH] Fix variable to get widget settings

This caused the panel to freeze when the cursor blink time was changed.
---
 plugins/launchtaskbar.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plugins/launchtaskbar.c b/plugins/launchtaskbar.c
index 3c9dc0b8..c239e73a 100644
--- a/plugins/launchtaskbar.c
+++ b/plugins/launchtaskbar.c
@@ -962,7 +962,7 @@ static void on_gtk_cursor_blink_time_changed(GObject *gsettings, GParamSpec *psp
     if (tb->flash_timeout == 0) /* nothing to do? */
         return;
     g_source_remove(tb->flash_timeout);
-    g_object_get(gtk_widget_get_settings(GTK_WIDGET(tb)), "gtk-cursor-blink-time",
+    g_object_get(gtk_widget_get_settings(GTK_WIDGET(tb->plugin)), "gtk-cursor-blink-time",
                  &interval, NULL);
     tb->flash_timeout = g_timeout_add(interval / 2, flash_window_timeout, tb);
 }
